### PR TITLE
Tabular: Fixed defect in transform_features + new arg

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*

- Fixed defect in transform_features when training data's indices are not 0-n
- Added return_original_features argument to simplify obtaining pure OOF predictions

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-517-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-517/2/index.html

**Comment by szha** [comment]
Job PR-517-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-517/3/index.html

File: autogluon/task/tabular_prediction/predictor.py [hunk 1]
```diff
@@ -412,6 +412,10 @@ def transform_features(self, dataset=None, model=None, base_models=None):
             Valid models are listed in this `predictor` by calling `predictor.model_names`.
             If a stacker model S exists with `base_models=M`, then setting `base_models=M` is equivalent to setting `model=S`.
             `model=None` is a requirement when specifying `base_models`.
+        return_original_features : bool, default = True
```
Comment (jwmueller) [inline autogluon/task/tabular_prediction/predictor.py:415]: `return_original_features` option could be clarified a bit more, eg by precisely stating what sort of DF will be returned when it is true vs when it is false, as well as describing for what purposes one might want it to be True vs False.

**Comment by Innixma** [comment]
> Just clarify the documentation in `transform_features`.
> 
> Also, why not have `use_orig_features` be an option of the stacker models that can be specified by user in task.fit() or optimized in HPO?

This is planned to be added in future

File: autogluon/task/tabular_prediction/predictor.py (hunk 1 above)
Comment (Innixma) [reply autogluon/task/tabular_prediction/predictor.py:415]: Will clarify in future PR

**Pull request merged**